pennylane/expval/__init__.py

```diff
@@ -50,5 +50,44 @@
     expval/cv
 """
 
-from .qubit import *
-from .cv import *
+from pennylane.qnode import QNode, QuantumFunctionError
+
+from .qubit import * #pylint: disable=unused-wildcard-import,wildcard-import
+from .cv import * #pylint: disable=unused-wildcard-import,wildcard-import
+
+from .cv import __all__ as _cv__all__
+from .qubit import __all__ as _qubit__all__
+
+class Identity(object): #pylint: disable=too-few-public-methods,function-redefined
+    r"""pennylane.expval.Identity(wires)
+    Expectation value of the identity observable :math:`\I`.
+
+    The expectation of this observable
+
+    .. math::
+        E[\I] = \text{Tr}(\I \rho)
+
+    corresponds to the trace of the quantum state, which in exact
+    simulators should always be equal to 1.
+
+    This is a placeholder for the Identity classes in expval.qubit and expval.cv
+    and instantiates the Identity appropriate for the respective device.
+    """
+    def __new__(cls, *args, **kwargs):
+        if QNode._current_context is None:
+            raise QuantumFunctionError("Quantum operations can only be used inside a qfunc.")
+
+        supported_expectations = QNode._current_context.device.expectations
+        if supported_expectations.intersection([cls for cls in _cv__all__]):
+            return cv.Identity(*args, **kwargs)
```

Comment:
I'm probably missing something, but I'm a bit confused why this line works? We are never explicitly importing `cv` or `qubit` into the namespace anywhere, just their contents

Reply:
The lines have changed now, so this is obsolete.